didx-xyz/yoma-mobile
# YOMA-721: refactor(Education):  enable saving education

## Scope [YOMA-721](https://yomaworld.atlassian.net/browse/YOMA-721)

![](https://media.giphy.com/media/Q8rwlNTcDAU3MuUzd7/giphy-downsized.gif)
## Work Done
_this is probably far too big of a PR_
There was a rather large amount of work completed. This is a summary:
1. Split education out of the Qualifications module. Same for UserEducation. Will refactor Qualifications into Learning Opportunities in the future. 
2. Got all views and widgets working to display the correct data (was a data mapping issue against the response data).
3. Had to update some of the API endpoints to cater for Education
4. There has grown a disconnect between the enums for credentials when we're creating credentials on the API and for when we're receiving credentials as data - so needed to have different enums for that. 
5. Did some other refactors as I worked. Also ran into build issues in the middle of the job (randomly) - implemented some changes on the Apex branch which was building fine which fixed the issue.

## Steps to Test
- run the app, create a new education credential. Should all work (except certificates- that's not working 100% correctly as yet)
